superset/security/manager.py

```diff
@@ -1047,17 +1048,23 @@ def raise_for_access(
 
             assert datasource
 
-            if not (
-                self.can_access_schema(datasource)
-                or self.can_access("datasource_access", datasource.perm or "")
-                or (
-                    feature_flag_manager.is_feature_enabled("DASHBOARD_RBAC")
-                    and self.can_access_based_on_dashboard(datasource)
-                )
-            ):
-                raise SupersetSecurityException(
-                    self.get_datasource_access_error_object(datasource)
-                )
+            self.raise_when_there_is_no_access_to(datasource)
+
+    def raise_when_there_is_no_access_to(self, datasource: BaseDatasource) -> None:
```

Comment:
This name should IMO follow the existing naming conventions. In this case `raise_for_datasource_access` would be more appropriate and more aligned with the pre-existing `raise_for_dashboard_access`.

Reply:
"raise_for_dashboard_acess" is even more bad name. what is mean "dashboard_acess" ?

a better name conventions are on of those: validate_<object_type>_can_be_accessed
or 
actor_can_access_<object_type>

Reply:
We can discuss which naming strategy is better, but I think we can all agree that consistency is a good thing. If we have an existing pattern I think it would be better to stick with it, since it makes it easier for people to understand the code.

Reply:
noting that this pattern was originally used (i think) because it follows the similar one used by the python requests package, `raise_for_status`: https://docs.python-requests.org/en/latest/api/#requests.Response.raise_for_status

@john-bodley might know a bit more about why we picked this naming

Reply:
@etr2460 
You gave a good reference, but I still think is an anti-pattern, and they are wrong too.
but with that reference, it should be discussed in a different context not in that PR

@betodealmeida 
There is a conflict between two principles: 
consistency vs you should always leave the code you have just modified cleaner than it was, a good catch and currently, I don't have a good argument or opinion on what principle is stronger. 

I'll rename it